crates/evm/src/evm/handler.rs

```diff
@@ -512,15 +527,17 @@ fn calc_diff_size<EXT, DB: Database>(
 
             // All the nonce, balance and code_hash fields are updated and written to the state with DbAccount
             diff_size +=
-                (DB_ACCOUNT_SIZE + DB_ACCOUNT_KEY_SIZE) * NONCE_DISCOUNTED_PERCENTAGE / 100;
+                (db_account_size + DB_ACCOUNT_KEY_SIZE) * NONCE_DISCOUNTED_PERCENTAGE / 100;
 
             // Retrieve code from DB and apply its size
             if let Some(info) = db.basic(*addr)? {
                 if let Some(code) = info.code {
                     diff_size += code.len();
```

Comment:
if we are destroting account, we shouldn't be charging code.len();

we should be charging dbaccount.keys.len() * (dbaccount.keys key size + dbaccount.storage keysize + 2 (for writing null) ) +